diff --git a/CMakeLists.txt b/CMakeLists.txt
index 35386379d6..59a099df70 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -135,7 +135,7 @@ if (UNIX AND NOT CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   endif ()
 endif ()
 
-add_subdirectory(ext)
+add_subdirectory(ext EXCLUDE_FROM_ALL)
 add_subdirectory(src)
 
 add_subdirectory(support/coding/codegen)
diff --git a/apps/OpenSpace-MinVR/CMakeLists.txt b/apps/OpenSpace-MinVR/CMakeLists.txt
index f924e8d1ea..c7d597d8bc 100644
--- a/apps/OpenSpace-MinVR/CMakeLists.txt
+++ b/apps/OpenSpace-MinVR/CMakeLists.txt
@@ -28,7 +28,7 @@ include(${PROJECT_SOURCE_DIR}/support/cmake/global_variables.cmake)
 set(MACOSX_BUNDLE_ICON_FILE openspace.icns)
 
 set(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT ON CACHE BOOL "" FORCE)
-add_subdirectory(ext/minvr)
+add_subdirectory(ext/minvr EXCLUDE_FROM_ALL)
 
 create_new_application(OpenSpace-MinVR
   ${CMAKE_CURRENT_SOURCE_DIR}/main.cpp
diff --git a/apps/OpenSpace/CMakeLists.txt b/apps/OpenSpace/CMakeLists.txt
index d6f99b9c5e..0f48b5c646 100644
--- a/apps/OpenSpace/CMakeLists.txt
+++ b/apps/OpenSpace/CMakeLists.txt
@@ -121,7 +121,7 @@ if (NOT OPENSPACE_HAVE_TESTS)
   # Unless overriden, disable sgct tests if OpenSpace tests are disabled
   set(SGCT_BUILD_TESTS FALSE CACHE BOOL "Enable ghoul unit tests")
 endif()
-add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/ext/sgct)
+add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/ext/sgct EXCLUDE_FROM_ALL)
 target_link_libraries(OpenSpace PRIVATE sgct)
 
 set_target_properties(sgct PROPERTIES FOLDER "External")
diff --git a/modules/audio/CMakeLists.txt b/modules/audio/CMakeLists.txt
index 93cc01e6b0..ac875083f2 100644
--- a/modules/audio/CMakeLists.txt
+++ b/modules/audio/CMakeLists.txt
@@ -49,7 +49,7 @@ if (WIN32)
 elseif (UNIX)
   set(SOLOUD_BACKEND_ALSA ON CACHE BOOL "")
 endif ()
-add_subdirectory(ext/soloud/contrib SYSTEM)
+add_subdirectory(ext/soloud/contrib SYSTEM EXCLUDE_FROM_ALL)
 
 # Unfortunately, the soloud cmake tarket doesn't set the include directories correctly
 target_include_directories(openspace-module-audio SYSTEM PRIVATE ext/soloud/include)
diff --git a/modules/fitsfilereader/CMakeLists.txt b/modules/fitsfilereader/CMakeLists.txt
index f1d0ffdc5e..8e770d029c 100644
--- a/modules/fitsfilereader/CMakeLists.txt
+++ b/modules/fitsfilereader/CMakeLists.txt
@@ -48,11 +48,11 @@ create_new_module(
 )
 
 # CCfits is dependent on cfitsio, let it handle the internal linking
-add_subdirectory(ext/cfitsio SYSTEM)
+add_subdirectory(ext/cfitsio SYSTEM EXCLUDE_FROM_ALL)
 set_target_properties(cfitsio PROPERTIES FOLDER "External")
 set(cfitsio_BUILD_SHARED_LIBS OFF)
 
-add_subdirectory(ext/CCfits SYSTEM)
+add_subdirectory(ext/CCfits SYSTEM EXCLUDE_FROM_ALL)
 set_target_properties(CCfits PROPERTIES FOLDER "External")
 set(CCfits_BUILD_SHARED_LIBS OFF)
 
diff --git a/modules/globebrowsing/CMakeLists.txt b/modules/globebrowsing/CMakeLists.txt
index 11fbeff462..8075d83e35 100644
--- a/modules/globebrowsing/CMakeLists.txt
+++ b/modules/globebrowsing/CMakeLists.txt
@@ -176,7 +176,7 @@ else (WIN32)
 endif () # WIN32
 
 begin_dependency("GEOS")
-add_subdirectory(ext/geos SYSTEM)
+add_subdirectory(ext/geos SYSTEM EXCLUDE_FROM_ALL)
 target_link_libraries(openspace-module-globebrowsing PRIVATE geos)
 
 if (MSVC)
diff --git a/modules/imgui/CMakeLists.txt b/modules/imgui/CMakeLists.txt
index e01ee545f8..6674bbdcc1 100644
--- a/modules/imgui/CMakeLists.txt
+++ b/modules/imgui/CMakeLists.txt
@@ -74,7 +74,7 @@ create_new_module(
   ${HEADER_FILES} ${SOURCE_FILES} ${SHADER_FILES}
 )
 
-add_subdirectory(ext/imgui SYSTEM)
+add_subdirectory(ext/imgui SYSTEM EXCLUDE_FROM_ALL)
 target_link_libraries(${imgui_module} PUBLIC Imgui)
 disable_compiler_warnings(Imgui)
 set_target_properties(Imgui PROPERTIES FOLDER "External")
diff --git a/modules/kameleon/CMakeLists.txt b/modules/kameleon/CMakeLists.txt
index 606dc81bf2..70910905ab 100644
--- a/modules/kameleon/CMakeLists.txt
+++ b/modules/kameleon/CMakeLists.txt
@@ -53,7 +53,7 @@ mark_as_advanced(BUILD_SHARED_LIBS) # Change to set instead of option
 option(KAMELEON_USE_HDF5 "Kameleon use HDF5" OFF)
 set(KAMELEON_ROOT_DIR ${CMAKE_CURRENT_SOURCE_DIR}/ext/kameleon)
 set(KAMELEON_INCLUDES ${KAMELEON_ROOT_DIR}/src)
-add_subdirectory(${KAMELEON_ROOT_DIR} SYSTEM)
+add_subdirectory(${KAMELEON_ROOT_DIR} SYSTEM EXCLUDE_FROM_ALL)
 target_include_directories(${kameleon_module} SYSTEM PUBLIC ${KAMELEON_INCLUDES})
 target_link_libraries(${kameleon_module} PRIVATE ccmc)
 mark_as_advanced(CDF_BUILD_ZLIB CDF_INCLUDES CDF_LIBRARY CDF_USE_STATIC_LIBS
