diff --git a/CMakeLists.txt b/CMakeLists.txt
index ec7b233c2..e5be18929 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -120,7 +120,7 @@ if (UNIX AND NOT CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   add_compile_definitions(D_GNU_SOURCE)
 endif ()
 
-add_subdirectory(ext)
+add_subdirectory(ext EXCLUDE_FROM_ALL)
 add_subdirectory(src)
 
 add_subdirectory(support/coding/codegen)
diff --git a/apps/OpenSpace-MinVR/CMakeLists.txt b/apps/OpenSpace-MinVR/CMakeLists.txt
index ef5cc6608..cec6de379 100644
--- a/apps/OpenSpace-MinVR/CMakeLists.txt
+++ b/apps/OpenSpace-MinVR/CMakeLists.txt
@@ -28,7 +28,7 @@ include(${PROJECT_SOURCE_DIR}/support/cmake/global_variables.cmake)
 set(MACOSX_BUNDLE_ICON_FILE openspace.icns)
 
 set(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT ON CACHE BOOL "" FORCE)
-add_subdirectory(ext/minvr)
+add_subdirectory(ext/minvr EXCLUDE_FROM_ALL)
 
 create_new_application(OpenSpace-MinVR
   ${CMAKE_CURRENT_SOURCE_DIR}/main.cpp
diff --git a/apps/OpenSpace/CMakeLists.txt b/apps/OpenSpace/CMakeLists.txt
index 8c4999803..2dd118284 100644
--- a/apps/OpenSpace/CMakeLists.txt
+++ b/apps/OpenSpace/CMakeLists.txt
@@ -112,7 +112,7 @@ if (NOT OPENSPACE_HAVE_TESTS)
   # Unless overriden, disable sgct tests if OpenSpace tests are disabled
   set(SGCT_BUILD_TESTS FALSE CACHE BOOL "Enable ghoul unit tests")
 endif()
-add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/ext/sgct)
+add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/ext/sgct EXCLUDE_FROM_ALL)
 target_link_libraries(OpenSpace PRIVATE sgct)
 
 set_target_properties(sgct PROPERTIES FOLDER "External")
diff --git a/modules/audio/CMakeLists.txt b/modules/audio/CMakeLists.txt
index 7daa81154..b5a180ad8 100644
--- a/modules/audio/CMakeLists.txt
+++ b/modules/audio/CMakeLists.txt
@@ -48,7 +48,7 @@ if (WIN32)
 elseif (UNIX)
   set(SOLOUD_BACKEND_ALSA ON CACHE BOOL "")
 endif ()
-add_subdirectory(ext/soloud/contrib SYSTEM)
+add_subdirectory(ext/soloud/contrib SYSTEM EXCLUDE_FROM_ALL)
 
 # Unfortunately, the soloud cmake tarket doesn't set the include directories correctly
 target_include_directories(openspace-module-audio SYSTEM PRIVATE ext/soloud/include)
diff --git a/modules/fitsfilereader/CMakeLists.txt b/modules/fitsfilereader/CMakeLists.txt
index b942246f3..a78ca3344 100644
--- a/modules/fitsfilereader/CMakeLists.txt
+++ b/modules/fitsfilereader/CMakeLists.txt
@@ -47,10 +47,10 @@ create_new_module(
 )
 
 # CCfits is dependent on cfitsio, let it handle the internal linking
-add_subdirectory(ext/cfitsio SYSTEM)
+add_subdirectory(ext/cfitsio SYSTEM EXCLUDE_FROM_ALL)
 set_target_properties(cfitsio PROPERTIES FOLDER "External")
 
-add_subdirectory(ext/CCfits SYSTEM)
+add_subdirectory(ext/CCfits SYSTEM EXCLUDE_FROM_ALL)
 set_target_properties(CCfits PROPERTIES FOLDER "External")
 
 if (MSVC)
diff --git a/modules/globebrowsing/CMakeLists.txt b/modules/globebrowsing/CMakeLists.txt
index a7bdcc5f8..8264853fc 100644
--- a/modules/globebrowsing/CMakeLists.txt
+++ b/modules/globebrowsing/CMakeLists.txt
@@ -177,7 +177,7 @@ else (WIN32)
 endif () # WIN32
 
 begin_dependency("GEOS")
-add_subdirectory(ext/geos SYSTEM)
+add_subdirectory(ext/geos SYSTEM EXCLUDE_FROM_ALL)
 target_link_libraries(openspace-module-globebrowsing PRIVATE geos)
 
 if (MSVC)
diff --git a/modules/imgui/CMakeLists.txt b/modules/imgui/CMakeLists.txt
index 098d895a0..22ba8e35e 100644
--- a/modules/imgui/CMakeLists.txt
+++ b/modules/imgui/CMakeLists.txt
@@ -75,7 +75,7 @@ create_new_module(
   ${HEADER_FILES} ${SOURCE_FILES} ${SHADER_FILES}
 )
 
-add_subdirectory(ext/imgui SYSTEM)
+add_subdirectory(ext/imgui SYSTEM EXCLUDE_FROM_ALL)
 target_link_libraries(${imgui_module} PUBLIC Imgui)
 disable_compiler_warnings(Imgui)
 set_target_properties(Imgui PROPERTIES FOLDER "External")
diff --git a/modules/kameleon/CMakeLists.txt b/modules/kameleon/CMakeLists.txt
index 53277a1e3..183d67552 100644
--- a/modules/kameleon/CMakeLists.txt
+++ b/modules/kameleon/CMakeLists.txt
@@ -46,7 +46,7 @@ option(KAMELEON_LIBRARY_ONLY "Build with Kameleon as library only" ON)
 option(KAMELEON_USE_HDF5 "Kameleon use HDF5" OFF)
 set(KAMELEON_ROOT_DIR ${CMAKE_CURRENT_SOURCE_DIR}/ext/kameleon)
 set(KAMELEON_INCLUDES ${KAMELEON_ROOT_DIR}/src)
-add_subdirectory(${KAMELEON_ROOT_DIR} SYSTEM)
+add_subdirectory(${KAMELEON_ROOT_DIR} SYSTEM EXCLUDE_FROM_ALL)
 target_include_directories(${kameleon_module} SYSTEM PUBLIC ${KAMELEON_INCLUDES})
 target_link_libraries(${kameleon_module} PRIVATE ccmc)
 mark_as_advanced(CDF_BUILD_ZLIB CDF_INCLUDES CDF_LIBRARY CDF_USE_STATIC_LIBS
