diff --git a/CMakeLists.txt b/CMakeLists.txt
index 268a7d9f9e..0da0db26bd 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -158,23 +158,6 @@ set_target_properties(codegen-tool PROPERTIES FOLDER "support")
 set_target_properties(run_codegen PROPERTIES FOLDER "support")
 
 
-begin_header("Configuring Modules")
+# Qt
+# Unfortunately, we have to set this value manually; sigh
+# In the future, if the Qt version is updated, just add to this variable ---abock
+if (APPLE)
+  set(CMAKE_PREFIX_PATH
+    "~/Qt/5.6/clang_64/lib/cmake"
+    "~/Qt/5.7/clang_64/lib/cmake"
+    "~/Qt/5.8/clang_64/lib/cmake"
+    "~/Qt/5.9/clang_64/lib/cmake"
+    "~/Qt/5.10/clang_64/lib/cmake"
+    "~/Qt/5.11/clang_64/lib/cmake"
+    "~/Qt/5.12/clang_64/lib/cmake"
+    "~/Qt/5.15.1/clang_64/lib/cmake"
+    "~/Qt/6.2.3/macos/lib/cmake"
+  )
+endif ()
+
+begin_header("Configuring Modules")
 add_subdirectory("${CMAKE_CURRENT_SOURCE_DIR}/modules")
 end_header("End: Configuring Modules")
diff --git a/apps/OpenSpace/CMakeLists.txt b/apps/OpenSpace/CMakeLists.txt
index 275ee8b16b..8c49998033 100644
--- a/apps/OpenSpace/CMakeLists.txt
+++ b/apps/OpenSpace/CMakeLists.txt
@@ -72,15 +72,6 @@ if (SGCT_SPOUT_SUPPORT AND NOT OPENSPACE_MODULE_SPOUT)
 endif ()
 
 
-create_new_application(OpenSpace
+#####
+# MacOS
+#####
+if (APPLE)
+  set(MACOSX_BUNDLE_ICON_FILE openspace.icns)
+  set(CMAKE_OSX_DEPLOYMENT_TARGET "13.3" CACHE STRING "Minimum OS X deployment version" FORCE)
+  set(CMAKE_XCODE_ATTRIBUTE_OTHER_CODE_SIGN_FLAGS "--deep" CACHE STRING "Other Code Signing Flags" FORCE)
+endif ()
+
+create_new_application(OpenSpace
   ${SGCT_OPENVR_FILES}
   ${CMAKE_CURRENT_SOURCE_DIR}/main.cpp
@@ -130,7 +121,7 @@ if (TARGET SGCTTest)
   set_target_properties(SGCTTest PROPERTIES FOLDER "External")
 endif ()
 
+if (UNIX AND (NOT APPLE))
-if (UNIX)
   target_link_libraries(OpenSpace PRIVATE Xcursor Xinerama X11)
 endif ()
 
diff --git a/apps/OpenSpace/main.cpp b/apps/OpenSpace/main.cpp
index 7d2d91e244..2c66c7893d 100644
--- a/apps/OpenSpace/main.cpp
+++ b/apps/OpenSpace/main.cpp
@@ -22,12 +22,6 @@
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                                         *
  ****************************************************************************************/
 
-#include <openspace/documentation/documentation.h>
+// Explicitly including OpenGL for APPLE
+#ifdef __APPLE__
+#define GLFW_INCLUDE_NONE
+#include <OpenGL/gl3.h>
+#endif // __APPLE__
+
+#include <openspace/documentation/documentation.h>
 #include <openspace/engine/configuration.h>
 #include <openspace/engine/globals.h>
@@ -335,8 +329,6 @@ void mainInitFunc(GLFWwindow*) {
     global::openSpaceEngine->initialize();
     LDEBUG("Initializing OpenSpace Engine finished");
 
-    {
+#ifndef __APPLE__
+    // Apparently: "Cocoa: Regular windows do not have icons on macOS"
+    {
         const std::filesystem::path path = absPath("${DATA}/openspace-icon.png");
         int x = 0;
@@ -356,7 +348,6 @@ void mainInitFunc(GLFWwindow*) {
 
         stbi_image_free(icon.pixels);
     }
-
+#endif // __APPLE__
+
     currentWindow = Engine::instance().windows().front().get();
     currentViewport = currentWindow->viewports().front().get();
@@ -1448,22 +1439,10 @@ int main(int argc, char* argv[]) {
 
     global::openSpaceEngine->createUserDirectoriesIfNecessary();
 
-    if (!global::configuration->bypassLauncher) {
+    // (abock, 2020-12-07)  For some reason on Apple the keyboard handler in CEF will call
+    // the Qt one even if the QApplication was destroyed, leading to invalid memory
+    // access.  The only way we could fix this for the release was to keep the
+    // QApplication object around until the end of the program.  Even though the Qt
+    // keyboard handler gets called, it doesn't do anything so everything still works.
+#ifdef __APPLE__
+    int qac = 0;
+    QApplication app(qac, nullptr);
+#endif // __APPLE__
+
+    if (!global::configuration->bypassLauncher) {
 #ifdef OPENSPACE_HAS_LAUNCHER
-        int qac = 0;
+#ifndef __APPLE__
+        int qac = 0;
         QApplication app(qac, nullptr);
-
+#endif // __APPLE__
+
         std::string pwd = std::filesystem::current_path().string();
         if (const size_t it = pwd.find_first_of("'\"[]");  it != std::string::npos) {
@@ -1614,10 +1593,6 @@ int main(int argc, char* argv[]) {
     Log::instance().setShowLogLevel(false);
     Log::instance().setLogCallback(mainLogCallback);
 
-    std::filesystem::path winConf =
+#ifdef __APPLE__
+    glfwWindowHint(GLFW_STENCIL_BITS, 8);
+#endif
+
+    std::filesystem::path winConf =
         commandlineArguments.windowConfig.has_value() ?
         *commandlineArguments.windowConfig :
@@ -1684,30 +1659,6 @@ int main(int argc, char* argv[]) {
         throw;
     }
 
-    LINFO("Starting rendering loop");
+#ifdef __APPLE__
+    // Workaround for OpenGL bug that Apple introduced in 10.14 Mojave that prevents an
+    // OpenGL context to display anything until it is first moved or resized in dark
+    // mode. So we are going through all windows here and resize them a bit larger and
+    // then back to the desired resolution. Resizing the window to the same size doesn't
+    // work as GLFW probably has a check for setting the current values.
+    // This can be removed once the OpenGL bug is fixed.
+    // In order to check, comment out the following lines and start OpenSpace on a 10.14
+    // machine. If the loading screen shows up without doing anything to the window, it
+    // is fixed. With the bug, the rendering stays gray even well after the main render
+    // loop has started     -- 2018-10-28   abock
+    for (const std::unique_ptr<Window>& window : Engine::instance().windows()) {
+        GLFWwindow* w = window->windowHandle();
+        int x, y;
+        glfwGetWindowPos(w, &x, &y);
+        glfwSetWindowPos(w, x + 1, y + 1);
+        glfwSwapBuffers(w);
+        glfwPollEvents();
+        glfwSetWindowPos(w, x, y);
+        glfwSwapBuffers(w);
+        glfwPollEvents();
+    }
+#endif // __APPLE__
+
+    LINFO("Starting rendering loop");
     Engine::instance().exec();
     LINFO("Ending rendering loop");
diff --git a/ext/CMakeLists.txt b/ext/CMakeLists.txt
index dd745db73d..6561633c5c 100644
--- a/ext/CMakeLists.txt
+++ b/ext/CMakeLists.txt
@@ -22,28 +22,6 @@
 # OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                                          #
 ##########################################################################################
 
-# Ghoul
+# System libraries
+if (APPLE)
+  begin_dependency("Core Libraries")
+  add_library(external-system-apple INTERFACE)
+
+  target_include_directories(external-system-apple INTERFACE "/Developer/Headers/FlatCarbon")
+  find_library(COREFOUNDATION_LIBRARY CoreFoundation REQUIRED)
+  find_library(CARBON_LIBRARY Carbon REQUIRED)
+  find_library(COCOA_LIBRARY Cocoa REQUIRED) # This was Carbon before
+  find_library(APP_SERVICES_LIBRARY ApplicationServices REQUIRED)
+  find_library(APPKIT_LIBRARY AppKit REQUIRED)
+  mark_as_advanced(COREFOUNDATION_LIBRARY CARBON_LIBRARY COCOA_LIBRARY
+    APP_SERVICES_LIBRARY APPKIT_LIBRARY
+  )
+  target_link_libraries(external-system-apple INTERFACE # This was PUBLIC before
+    ${CARBON_LIBRARY} ${COREFOUNDATION_LIBRARY}
+    ${COCOA_LIBRARY} ${APP_SERVICES_LIBRARY}
+    ${APPKIT_LIBRARY}
+  )
+  end_dependency()
+endif ()
+
+# Ghoul
 if (NOT OPENSPACE_HAVE_TESTS)
   # Unless overriden, disable ghoul tests if OpenSpace tests are disabled
diff --git a/modules/audio/CMakeLists.txt b/modules/audio/CMakeLists.txt
index 7ec100055c..24ee7798af 100644
--- a/modules/audio/CMakeLists.txt
+++ b/modules/audio/CMakeLists.txt
@@ -46,10 +46,8 @@ begin_dependency("SoLoud")
 set(SOLOUD_BACKEND_SDL2 OFF CACHE BOOL "")
 if (WIN32)
   set(SOLOUD_BACKEND_WINMM ON CACHE BOOL "")
+elseif (UNIX AND NOT APPLE)
-elseif (UNIX)
   set(SOLOUD_BACKEND_ALSA ON CACHE BOOL "")
+elseif (APPLE)
+  set(SOLOUD_BACKEND_COREAUDIO ON CACHE BOOL "")
 endif ()
 add_subdirectory(ext/soloud/contrib SYSTEM)
 
diff --git a/modules/base/rendering/grids/renderableboxgrid.cpp b/modules/base/rendering/grids/renderableboxgrid.cpp
index 6725c47c77..1f9cce9b6a 100644
--- a/modules/base/rendering/grids/renderableboxgrid.cpp
+++ b/modules/base/rendering/grids/renderableboxgrid.cpp
@@ -155,11 +155,7 @@ void RenderableBoxGrid::render(const RenderData& data, RendererTasks&) {
     _gridProgram->setUniform("gridColor", _color);
 
     // Change GL state:
-    glLineWidth(_lineWidth);
+#ifndef __APPLE__
+    glLineWidth(_lineWidth);
-    glBlendFunc(GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA);
+#else // ^^^^ __APPLE__ // !__APPLE__ vvvv
+    glLineWidth(1.f);
+#endif // __APPLE__
+    glBlendFunc(GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA);
     glEnablei(GL_BLEND, 0);
     glEnable(GL_LINE_SMOOTH);
diff --git a/modules/base/rendering/grids/renderablegrid.cpp b/modules/base/rendering/grids/renderablegrid.cpp
index 13acbf8302..4f0801222f 100644
--- a/modules/base/rendering/grids/renderablegrid.cpp
+++ b/modules/base/rendering/grids/renderablegrid.cpp
@@ -270,11 +270,7 @@ void RenderableGrid::render(const RenderData& data, RendererTasks&) {
     _gridProgram->setUniform("gridColor", _color);
 
     // Change GL state:
-    glLineWidth(_lineWidth);
+#ifndef __APPLE__
+    glLineWidth(_lineWidth);
-    glEnablei(GL_BLEND, 0);
+#else // ^^^^ __APPLE__ // !__APPLE__ vvvv
+    glLineWidth(1.f);
+#endif // __APPLE__
+    glEnablei(GL_BLEND, 0);
     glBlendFunc(GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA);
     glEnable(GL_LINE_SMOOTH);
@@ -285,11 +281,7 @@ void RenderableGrid::render(const RenderData& data, RendererTasks&) {
     glDrawArrays(_mode, 0, static_cast<GLsizei>(_varray.size()));
 
     // Render major grid
-    glLineWidth(_highlightLineWidth);
+#ifndef __APPLE__
+    glLineWidth(_highlightLineWidth);
-    _gridProgram->setUniform("gridColor", _highlightColor);
+#else // ^^^^ __APPLE__ // !__APPLE__ vvvv
+    glLineWidth(1.f);
+#endif // __APPLE__
+    _gridProgram->setUniform("gridColor", _highlightColor);
 
     glBindVertexArray(_highlightVao);
diff --git a/modules/base/rendering/grids/renderableradialgrid.cpp b/modules/base/rendering/grids/renderableradialgrid.cpp
index c37daa62e3..a3a368d475 100644
--- a/modules/base/rendering/grids/renderableradialgrid.cpp
+++ b/modules/base/rendering/grids/renderableradialgrid.cpp
@@ -210,11 +210,7 @@ void RenderableRadialGrid::render(const RenderData& data, RendererTasks&) {
     _gridProgram->setUniform("gridColor", _color);
 
     // Change GL state:
-    glLineWidth(_lineWidth);
+#ifndef __APPLE__
+    glLineWidth(_lineWidth);
-    glEnablei(GL_BLEND, 0);
+#else
+    glLineWidth(1.f);
+#endif
+    glEnablei(GL_BLEND, 0);
     glBlendFunc(GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA);
     glEnable(GL_LINE_SMOOTH);
diff --git a/modules/base/rendering/grids/renderablesphericalgrid.cpp b/modules/base/rendering/grids/renderablesphericalgrid.cpp
index b1059c11a0..f1d813ec07 100644
--- a/modules/base/rendering/grids/renderablesphericalgrid.cpp
+++ b/modules/base/rendering/grids/renderablesphericalgrid.cpp
@@ -219,12 +219,7 @@ void RenderableSphericalGrid::render(const RenderData& data, RendererTasks&) {
     _gridProgram->setUniform("gridColor", _color);
 
     // Change GL state:
-    glLineWidth(_lineWidth);
+#ifndef __APPLE__
+    glLineWidth(_lineWidth);
-    glEnablei(GL_BLEND, 0);
+#else // ^^^^ !__APPLE__ // __APPLE__ vvvv
+    glLineWidth(1.f);
+#endif // __APPLE__
+
+    glEnablei(GL_BLEND, 0);
     glBlendFunc(GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA);
     glEnable(GL_LINE_SMOOTH);
diff --git a/modules/base/rendering/renderablecartesianaxes.cpp b/modules/base/rendering/renderablecartesianaxes.cpp
index 3255e0f4ba..ed32b5c571 100644
--- a/modules/base/rendering/renderablecartesianaxes.cpp
+++ b/modules/base/rendering/renderablecartesianaxes.cpp
@@ -189,11 +189,7 @@ void RenderableCartesianAxes::render(const RenderData& data, RendererTasks&) {
     glBlendFunc(GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA);
     glEnablei(GL_BLEND, 0);
     glEnable(GL_LINE_SMOOTH);
-    glLineWidth(3.f);
+#ifndef __APPLE__
+    glLineWidth(3.f);
-
+#else // ^^^^ __APPLE__ // !__APPLE__ vvvv
+    glLineWidth(1.f);
+#endif // __APPLE__
+
     glBindVertexArray(_vao);
     glDrawElements(GL_LINES, 6, GL_UNSIGNED_INT, nullptr);
diff --git a/modules/base/rendering/renderabletrail.cpp b/modules/base/rendering/renderabletrail.cpp
index 59e57109c0..3c565bee63 100644
--- a/modules/base/rendering/renderabletrail.cpp
+++ b/modules/base/rendering/renderabletrail.cpp
@@ -44,21 +44,12 @@
 #include <optional>
 
 namespace {
-    constexpr std::array<const char*, 18> UniformNames = {
+#ifdef __APPLE__
+    constexpr std::array<const char*, 16> UniformNames = {
+        "opacity", "modelViewTransform", "projectionTransform", "color", "useLineFade",
+        "lineLength", "lineFadeAmount", "vertexSortingMethod", "idOffset", "nVertices",
+        "stride", "pointSize", "renderPhase", "useSplitRenderMode", "floatingOffset",
+        "numberOfUniqueVertices"
+    };
+#else // ^^^^ __APPLE__ // !__APPLE__ vvvv
+    constexpr std::array<const char*, 18> UniformNames = {
         "opacity", "modelViewTransform", "projectionTransform", "color", "useLineFade",
         "lineLength", "lineFadeAmount", "vertexSortingMethod", "idOffset", "nVertices",
         "stride", "pointSize", "renderPhase", "viewport", "lineWidth", "floatingOffset",
         "useSplitRenderMode", "numberOfUniqueVertices"
     };
-
+#endif // __APPLE__
+
     // The possible values for the _renderingModes property
     enum RenderingMode {
@@ -259,18 +250,6 @@ void RenderableTrail::initialize() {
 void RenderableTrail::initializeGL() {
     ZoneScoped;
 
-    _programObject = BaseModule::ProgramObjectManager.request(
+#ifdef __APPLE__
+    _programObject = BaseModule::ProgramObjectManager.request(
+        "EphemerisProgram",
+        []() -> std::unique_ptr<ghoul::opengl::ProgramObject> {
+            return global::renderEngine->buildRenderProgram(
+                "EphemerisProgram",
+                absPath("${MODULE_BASE}/shaders/renderabletrail_apple_vs.glsl"),
+                absPath("${MODULE_BASE}/shaders/renderabletrail_apple_fs.glsl")
+            );
+        }
+    );
+#else // ^^^^ __APPLE__ // !__APPLE__ vvvv
+    _programObject = BaseModule::ProgramObjectManager.request(
         "EphemerisProgram",
         []() -> std::unique_ptr<ghoul::opengl::ProgramObject> {
@@ -281,7 +260,6 @@ void RenderableTrail::initializeGL() {
             );
         }
     );
-
+#endif // __APPLE__
+
     ghoul::opengl::updateUniformLocations(*_programObject, _uniformCache, UniformNames);
 }
@@ -353,7 +331,6 @@ void RenderableTrail::internalRender(bool renderLines, bool renderPoints,
     _programObject->setUniform(_uniformCache.numberOfUniqueVertices,
         numberOfUniqueVertices);
 
-    std::array<GLint, 4> viewport;
+#ifndef __APPLE__
+    std::array<GLint, 4> viewport;
     global::renderEngine->openglStateCache().viewport(viewport.data());
     _programObject->setUniform(
@@ -367,7 +344,6 @@ void RenderableTrail::internalRender(bool renderLines, bool renderPoints,
         _uniformCache.lineWidth,
         std::ceil((2.f * 1.f + _appearance.lineWidth) * std::sqrt(2.f))
     );
-
+#endif // __APPLE__
+
     if (renderPoints) {
         // The stride parameter determines the distance between larger points and
@@ -460,11 +436,7 @@ void RenderableTrail::render(const RenderData& data, RendererTasks&) {
                               (_appearance.renderingModes == RenderingModeLinesPoints);
 
     if (renderLines) {
-        glLineWidth(std::ceil((2.f * 1.f + _appearance.lineWidth) * std::sqrt(2.f)));
+#ifdef __APPLE__
+        glLineWidth(1);
+#else // ^^^^ __APPLE__ // !__APPLE__ vvvv
+        glLineWidth(std::ceil((2.f * 1.f + _appearance.lineWidth) * std::sqrt(2.f)));
-    }
+#endif // __APPLE__
+    }
     if (renderPoints) {
         glEnable(GL_PROGRAM_POINT_SIZE);
diff --git a/modules/base/rendering/renderabletrail.h b/modules/base/rendering/renderabletrail.h
index caded421c0..3567f33811 100644
--- a/modules/base/rendering/renderabletrail.h
+++ b/modules/base/rendering/renderabletrail.h
@@ -193,18 +193,11 @@ class RenderableTrail : public Renderable {
 
     /// Program object used to render the data stored in RenderInformation
     ghoul::opengl::ProgramObject* _programObject = nullptr;
-    UniformCache(opacity, modelViewTransform, projectionTransform, color, useLineFade,
+#ifdef __APPLE__
+    UniformCache(opacity, modelViewTransform, projectionTransform, color, useLineFade,
+        lineLength, lineFadeAmount, vertexSortingMethod, idOffset, nVertices, stride,
+        pointSize, renderPhase, useSplitRenderMode, floatingOffset, numberOfUniqueVertices
+    ) _uniformCache;
+#else
+    UniformCache(opacity, modelViewTransform, projectionTransform, color, useLineFade,
         lineLength, lineFadeAmount, vertexSortingMethod, idOffset, nVertices, stride,
         pointSize, renderPhase, viewport, lineWidth, floatingOffset, useSplitRenderMode,
         numberOfUniqueVertices
     ) _uniformCache;
-};
+#endif
+};
 
 } // namespace openspace
diff --git a/modules/fieldlinessequence/rendering/renderablefieldlinessequence.cpp b/modules/fieldlinessequence/rendering/renderablefieldlinessequence.cpp
index fb1ebdacc6..3d7688456c 100644
--- a/modules/fieldlinessequence/rendering/renderablefieldlinessequence.cpp
+++ b/modules/fieldlinessequence/rendering/renderablefieldlinessequence.cpp
@@ -1295,11 +1295,7 @@ void RenderableFieldlinesSequence::render(const RenderData& data, RendererTasks&
     }
 
     glBindVertexArray(_vao);
-    glLineWidth(_lineWidth);
+#ifndef __APPLE__
+    glLineWidth(_lineWidth);
-
+#else
+    glLineWidth(1.f);
+#endif
+
     int loadedIndex = _activeIndex;
     if (loadedIndex == -1) {
diff --git a/modules/fitsfilereader/CMakeLists.txt b/modules/fitsfilereader/CMakeLists.txt
index 578a8265a5..8b5c6cbbf5 100644
--- a/modules/fitsfilereader/CMakeLists.txt
+++ b/modules/fitsfilereader/CMakeLists.txt
@@ -63,9 +63,6 @@ if (MSVC)
   target_compile_options(CCfits PRIVATE "/W0")
   target_compile_definitions(CCfits PRIVATE "_SCL_SECURE_NO_WARNINGS")
 else ()
-  target_compile_options(cfitsio PRIVATE "-w")
+  if (APPLE)
+    target_compile_options(cfitsio PRIVATE "-Wno-implicit-function-declaration")
+  endif ()
+  target_compile_options(cfitsio PRIVATE "-w")
   target_compile_options(CCfits PRIVATE "-w")
 endif ()

diff --git a/modules/globebrowsing/src/geojson/globegeometryfeature.cpp b/modules/globebrowsing/src/geojson/globegeometryfeature.cpp
index 05e1107c0c..dd59aab676 100644
--- a/modules/globebrowsing/src/geojson/globegeometryfeature.cpp
+++ b/modules/globebrowsing/src/geojson/globegeometryfeature.cpp
@@ -304,11 +304,7 @@ void GlobeGeometryFeature::render(const RenderData& renderData, int pass,
 
     const glm::dmat4 projectionTransform = renderData.camera.projectionMatrix();
 
-    glLineWidth(_properties.lineWidth() * extraRenderData.lineWidthScale);
+#ifndef __APPLE__
+    glLineWidth(_properties.lineWidth() * extraRenderData.lineWidthScale);
-
+#else  // ^^^^ __APPLE__ // !__APPLE__ vvvv
+    glLineWidth(1.f);
+#endif // __APPLE__
+
     for (const RenderFeature& r : _renderFeatures) {
         if (r.isExtrusionFeature && !_properties.extrude()) {
diff --git a/modules/globebrowsing/src/renderableglobe.cpp b/modules/globebrowsing/src/renderableglobe.cpp
index 76ab9b124b..7ff9647814 100644
--- a/modules/globebrowsing/src/renderableglobe.cpp
+++ b/modules/globebrowsing/src/renderableglobe.cpp
@@ -377,23 +377,23 @@ const Chunk& findChunkNode(const Chunk& node, const Geodetic2& location) {
     return *n;
 }
 
+#if defined(__APPLE__) || (defined(__linux__) && defined(__clang__))
-#if defined(__linux__) && defined(__clang__)
 using ChunkTileVector = std::vector<std::pair<ChunkTile, const LayerRenderSettings*>>;
 
 ChunkTileVector tilesAndSettingsUnsorted(const LayerGroup& layerGroup,
                                          const TileIndex& tileIndex)
 {
     ZoneScoped;

@@ a few lines down @@
+#if defined(__APPLE__) || (defined(__linux__) && defined(__clang__))
-#if (defined(__linux__) && defined(__clang__))

diff --git a/modules/space/rendering/renderabletravelspeed.cpp b/modules/space/rendering/renderabletravelspeed.cpp
index 76e91c09a3..185facf84d 100644
--- a/modules/space/rendering/renderabletravelspeed.cpp
+++ b/modules/space/rendering/renderabletravelspeed.cpp
@@ -319,11 +319,7 @@ void RenderableTravelSpeed::render(const RenderData& data, RendererTasks&) {
     _shaderProgram->setUniform("lineColor", _lineColor);
     _shaderProgram->setUniform("opacity", opacity());
 
-    glLineWidth(_lineWidth);
+#ifndef __APPLE__
+    glLineWidth(_lineWidth);
-    glBindVertexArray(_vao);
+#else // ^^^^ __APPLE__ // !__APPLE__ vvvv
+    glLineWidth(1.f);
+#endif // __APPLE__
+    glBindVertexArray(_vao);
     glDrawArrays(GL_LINE_STRIP, 0, 3);
     glBindVertexArray(0);
diff --git a/modules/webbrowser/src/cefhost.cpp b/modules/webbrowser/src/cefhost.cpp
index 4f5ffcb1de..accb22af85 100644
--- a/modules/webbrowser/src/cefhost.cpp
+++ b/modules/webbrowser/src/cefhost.cpp
@@ -33,10 +33,6 @@
 #include <filesystem>
 #include <string_view>
 
-#ifdef WIN32
+#ifdef __APPLE__
+#include <include/wrapper/cef_library_loader.h>
+#endif // __APPLE__
+
+#ifdef WIN32
 #pragma warning(push)
 #pragma warning(disable : 4100)
@@ -72,11 +68,7 @@ CefHost::CefHost([[maybe_unused]] const std::string& helperLocation) {
         std::filesystem::path(helperLocation).parent_path();
     const std::filesystem::path cefcache = std::format("{}/cefcache", root);
     CefString(&settings.root_cache_path).FromString(cefcache.string());
-    CefString(&settings.browser_subprocess_path).FromString(helperLocation);
+
+#ifndef __APPLE__
+    // Apple will always look for helper in a fixed location
+    CefString(&settings.browser_subprocess_path).FromString(helperLocation);
-
+#endif // __APPLE__
+
     settings.windowless_rendering_enabled = 1;
 
@@ -89,15 +81,6 @@ CefHost::CefHost([[maybe_unused]] const std::string& helperLocation) {
     // cf. https://github.com/chromiumembedded/cef/issues/3685
     settings.chrome_runtime = 1;
 
-    const CefRefPtr<WebBrowserApp> app = CefRefPtr<WebBrowserApp>(new WebBrowserApp);
+#ifdef __APPLE__
+    // Load the CEF framework library at runtime instead of linking directly as required
+    // by the macOS sandbox implementation
+    CefScopedLibraryLoader library_loader;
+    if (!library_loader.LoadInMain()) {
+        return;
+    }
+#endif // __APPLE__
+
+    const CefRefPtr<WebBrowserApp> app = CefRefPtr<WebBrowserApp>(new WebBrowserApp);
 
     const CefMainArgs args;
diff --git a/modules/webbrowser/webbrowsermodule.cpp b/modules/webbrowser/webbrowsermodule.cpp
index 10fdb30bb8..b5e01a1bc1 100644
--- a/modules/webbrowser/webbrowsermodule.cpp
+++ b/modules/webbrowser/webbrowsermodule.cpp
@@ -73,12 +73,9 @@ namespace {
     std::filesystem::path findHelperExecutable() {
 #ifdef WIN32
         constexpr std::string_view SubprocessPath = "OpenSpace_Helper.exe";
+#elif defined(__APPLE__)
+        constexpr std::string_view SubprocessPath =
+            "../Frameworks/OpenSpace Helper.app/Contents/MacOS/OpenSpace Helper";
+#else
-#else // ^^^^ WIN32 // !WIN32 vvvv
         constexpr std::string_view SubprocessPath = "OpenSpace_Helper";
+#endif
-#endif // WIN32
 
         const std::filesystem::path execLocation = absPath(std::format(
             "${{BIN}}/{}", SubprocessPath
@@ -245,7 +242,7 @@ bool WebBrowserModule::isEnabled() const {
 }
 
 bool WebBrowserModule::canUseAcceleratedRendering() {
+// Linux doesn't have a problem with the rendering, Apple doesn't support OpenGL 4.5
-// Linux doesn't have a problem with the rendering
 #ifdef WIN32
     ghoul::systemcapabilities::Version acceleratedVersion = {
         .major = 4, .minor = 5, .release = 0
diff --git a/modules/webgui/cmake/nodejs_support.cmake b/modules/webgui/cmake/nodejs_support.cmake
index 407169f246..067d9bc3fb 100644
--- a/modules/webgui/cmake/nodejs_support.cmake
+++ b/modules/webgui/cmake/nodejs_support.cmake
@@ -37,16 +37,6 @@ function(DownloadNodeJs version download_dir)
       set(filename "${basename}.exe")
       set(path "v${version}/win-x64/${filename}")
     endif ()
-  elseif (UNIX)
+  elseif (APPLE)
+    if (CMAKE_SYSTEM_PROCESSOR MATCHES "ARM64|aarch64")
+      set(basename "node-v${version}-darwin-arm64")
+      set(filename "${basename}.tar.gz")
+      set(path "v${version}/${filename}")
+    else () # Default to x64
+      set(basename "node-v${version}-darwin-x64")
+      set(filename "${basename}.tar.gz")
+      set(path "v${version}/${filename}")
+    endif ()
+  elseif (UNIX)
     if (CMAKE_SYSTEM_PROCESSOR MATCHES "ARM64|aarch64")
       set(basename "node-v${version}-linux-arm64")
@@ -81,15 +71,7 @@ function(DownloadNodeJs version download_dir)
   message(STATUS "URL: ${NODEJS_DOWNLOAD_URL}")
 
   # Extract the binary distribution for unix
+  if (APPLE)
+    # Apple uses tar.gz
+    message(STATUS "Extracting Node.js: ${NODEJS_DOWNLOAD_PATH} in ${download_dir}")
+    execute_process(
+      COMMAND tar xzf ${NODEJS_DOWNLOAD_PATH}
+      WORKING_DIRECTORY ${download_dir}
+    )
+  endif ()
+  if (UNIX AND NOT APPLE)
-  if (UNIX)
     # Linux uses tar.xz
     message(STATUS "Extracting Node.js: ${NODEJS_DOWNLOAD_PATH} in ${download_dir}")
     execute_process(
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 08a996273a..9e5eb53592 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -212,17 +212,6 @@ set(OPENSPACE_SOURCE
   util/versionchecker.cpp
 )
 
-set(OPENSPACE_HEADER
+if (APPLE)
+  set(OPENSPACE_SOURCE
+    ${OPENSPACE_SOURCE}
+    interaction/touchbar.mm
+  )
+  set_source_files_properties(
+    interaction/touchbar.mm
+    PROPERTIES SKIP_PRECOMPILE_HEADERS ON
+  )
+endif ()
+
+set(OPENSPACE_HEADER
   ${PROJECT_SOURCE_DIR}/include/openspace/json.h
   ${PROJECT_SOURCE_DIR}/include/openspace/camera/camera.h
@@ -423,13 +412,6 @@ set(OPENSPACE_HEADER
   ${PROJECT_SOURCE_DIR}/include/openspace/util/histogram.h
 )
 
-set(OPENSPACE_SHADERS
+if (APPLE)
+  set(OPENSPACE_HEADER
+    ${OPENSPACE_HEADER}
+    ${PROJECT_SOURCE_DIR}/include/openspace/interaction/touchbar.h
+  )
+endif ()
+
+set(OPENSPACE_SHADERS
   ${PROJECT_SOURCE_DIR}/shaders/blending.glsl
   ${PROJECT_SOURCE_DIR}/shaders/core/screenspace_fs.glsl
@@ -566,9 +548,6 @@ configure_file(
   @ONLY
 )
 
-if (WIN32)
+if (APPLE)
+  target_link_libraries(openspace-core INTERFACE external-system-apple)
+endif ()
+if (WIN32)
   target_link_libraries(openspace-core PUBLIC "pdh")
 endif ()
diff --git a/src/engine/openspaceengine.cpp b/src/engine/openspaceengine.cpp
index dbb5507fe2..2d5c6458a2 100644
--- a/src/engine/openspaceengine.cpp
+++ b/src/engine/openspaceengine.cpp
@@ -110,10 +110,6 @@
 #include "Psapi.h"
 #endif // WIN32
 
-#include "openspaceengine_lua.inl"
+#ifdef __APPLE__
+#include <openspace/interaction/touchbar.h>
+#endif // __APPLE__
+
+#include "openspaceengine_lua.inl"
 
 namespace {
@@ -884,10 +880,6 @@ void OpenSpaceEngine::loadAssets() {
         global::syncEngine->addSyncables(_scene->camera()->syncables());
     }
 
-    runGlobalCustomizationScripts();
+#ifdef __APPLE__
+    showTouchbar();
+#endif // APPLE
+
+    runGlobalCustomizationScripts();
 
     _writeDocumentationTask = std::async(
diff --git a/src/scene/scene_lua.inl b/src/scene/scene_lua.inl
index b0ff87034c..9821ce7ee2 100644
--- a/src/scene/scene_lua.inl
+++ b/src/scene/scene_lua.inl
@@ -308,9 +308,7 @@ std::vector<openspace::properties::Property*> findMatchesInAllProperties(
 
     std::mutex mutex;
     std::for_each(
-        std::execution::par_unseq,
+#ifndef __APPLE__
+        std::execution::par_unseq,
-        properties.cbegin(),
+#endif // __APPLE__
+        properties.cbegin(),
         properties.cend(),
         [&](Property* prop) {
@@ -358,9 +356,7 @@ std::vector<openspace::properties::PropertyOwner*> findMatchesInAllPropertyOwner
 
     std::mutex mutex;
     std::for_each(
-        std::execution::par_unseq,
+#ifndef __APPLE__
+        std::execution::par_unseq,
-        propertyOwners.cbegin(),
+#endif // __APPLE__
+        propertyOwners.cbegin(),
         propertyOwners.cend(),
         [&](PropertyOwner* propOwner) {

diff --git a/modules/solarbrowsing/util/solarbrowsinghelper.cpp b/modules/solarbrowsing/util/solarbrowsinghelper.cpp
index b0ff87034c..9821ce7ee2 100644
--- a/modules/solarbrowsing/util/solarbrowsinghelper.cpp
+++ b/modules/solarbrowsing/util/solarbrowsinghelper.cpp
@@ around line 650 @@
-    std::for_each(
-        std::execution::par,
+    std::for_each(
+#ifndef __APPLE__
+        std::execution::par,
+#endif // __APPLE__
